www/js/survey/input-matcher.ts

```diff
@@ -0,0 +1,243 @@
+import { logDebug, displayErrorMsg } from "../plugin/logger"
+import { DateTime } from "luxon";
+
+export type LocalDt = {
+    minute: number,
+    hour: number,
+    second: number,
+    day: number,
+    weekday: number,
+    month: number,
+    year: number,
+    timezone: string
+}
+
+export type UserInputForTrip = {
+    data: {
+        end_ts: number,
+        start_ts: number
+        label: string,
+        start_local_dt?: LocalDt
+        end_local_dt?: LocalDt
+        status?: string,
+        match_id?: string
+    },
+    metadata: {
+        time_zone: string,
+        plugin: string,
+        write_ts: number,
+        platform: string,
+        read_ts: number,
+        key: string
+    },
+    key?: string,
+}
+
+export type Trip = {
+    end_ts: number,
+    start_ts: number
+}
+
+export type TlEntry = {
+    key: string,
+    origin_key: string,
+    start_ts: number,
+    end_ts: number,
+    enter_ts: number,
+    exit_ts: number,
+    duration: number,
+    getNextEntry: any
+}
```

Comment:
It is so great that you're adding type definitions on all of these!

Having types like these will be useful in a lot of places, so I was hoping to keep common type definitions in a central place. Could you refer to `js/diary/diaryTypes.ts`? The types there are still incomplete, but you can fill in what you have here, and then import those types into this file.

This way, we will not have type definitions cluttering up every file.

Reply:
That sounds good to me. Go ahead and move `js/diary/diaryTypes.ts` into `js/types` and then expand on `diaryTypes.ts` with the types you use here.

Reply:
Updated it. Please check this commit: ffcc871d